core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -29,6 +30,16 @@ import org.apache.spark.serializer.{SerializerInstance, Serializer}
 import org.apache.spark.util.{CompletionIterator, Utils}
 
 /**
+ * Factory class that creates iterators to read records from fetched blocks
+ */
+trait BlockRecordIteratorFactory {
```

Comment:
Is this class intended to be public?  It might be okay as a developer API but I don't think that we should commit to making any of the shuffle internals as stable public interfaces.